src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallerFactory.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+
+namespace Microsoft.DotNet.Workloads.Workload.Install
+{
+    internal class WorkloadInstallerFactory
+    {
+        public static IWorkloadInstaller GetWorkloadInstaller(IReporter reporter)
```

Comment:
same here, if constructor works, use constructor

Reply:
I added this factory for when we have other types of workload installers. For now, since this the managed installer is the only one implemented, I wasn't sure if we wanted to add logic to error here if the SDK managed installer isn't appropriate (for example, with global installs that need MSIs)

Reply:
We would eventually need something to decide which installer to instantiate. Wouldn't hurt to have something here that we can iterate over as new installers become available.